server/models/Collective.js

```diff
@@ -2469,34 +2470,65 @@ export default function (Sequelize, DataTypes) {
       return Promise.resolve(null);
     }
 
-    const result = await models.Transaction.findOne({
-      attributes: [[Sequelize.fn('COALESCE', Sequelize.fn('SUM', Sequelize.col('Order.totalAmount')), 0), 'total']],
-      where: { HostCollectiveId: this.id, type: 'CREDIT' },
-      include: [
-        {
-          model: models.Order,
-          attributes: [],
-          where: { status: 'PAID' },
-          include: [
-            {
-              model: models.PaymentMethod,
-              as: 'paymentMethod',
-              attributes: [],
-              // This is the main chracateristic of Added Funds
-              // Some older usage before 2017 doesn't have it but it's ok
-              where: {
-                service: 'opencollective',
-                type: 'collective',
-                CollectiveId: this.id,
-              },
+    const include = [
+      {
+        model: models.Order,
+        attributes: [],
+        where: { status: 'PAID' },
+        include: [
+          {
+            model: models.PaymentMethod,
+            as: 'paymentMethod',
+            attributes: [],
+            // This is the main chracateristic of Added Funds
+            // Some older usage before 2017 doesn't have it but it's ok
+            where: {
+              service: 'opencollective',
+              type: 'collective',
+              CollectiveId: this.id,
             },
-          ],
+          },
+        ],
+      },
+    ];
+
+    const [usd, usdHost, otherCurrencies] = await Promise.all([
+      models.Transaction.findOne({
+        attributes: [[Sequelize.fn('COALESCE', Sequelize.fn('SUM', Sequelize.col('Order.totalAmount')), 0), 'total']],
+        where: { HostCollectiveId: this.id, type: 'CREDIT', currency: 'USD' },
+        include,
+        raw: true,
+      }),
+      models.Transaction.findOne({
+        attributes: [
+          [Sequelize.fn('COALESCE', Sequelize.fn('SUM', Sequelize.col('amountInHostCurrency')), 0), 'total'],
+        ],
+        where: { HostCollectiveId: this.id, type: 'CREDIT', currency: { [Op.ne]: 'USD' }, hostCurrency: 'USD' },
+        include,
+        raw: true,
+      }),
+      models.Transaction.findAll({
+        attributes: [
+          [Sequelize.col('Transaction.currency'), 'currency'],
+          [Sequelize.fn('COALESCE', Sequelize.fn('SUM', Sequelize.col('amount')), 0), 'total'],
+        ],
+        group: [Sequelize.col('Transaction.currency')],
+        where: {
+          HostCollectiveId: this.id,
+          type: 'CREDIT',
+          currency: { [Op.ne]: 'USD' },
+          hostCurrency: { [Op.ne]: 'USD' },
         },
-      ],
-      raw: true,
-    });
+        include,
+        raw: true,
+      }),
+    ]);
```

Comment:
I understand what's attempted, but I think it would be clearer to get the sum for all currencies in one SQL query, then proceed with the rates. Especially that the complex code has to be repeated multiple times.

We don't even need a special case for USD, because getFxRate knows it just need to skip conversion if USD -> USD.

Reply:
Cool, I simplified it, but it brings a bit of complexity to JS runtime, check it out: 0c34197.